File: ruby/high-scores/high_scores.rb
class HighScores
  @scores = []

  attr_accessor :scores

  def initialize(scores)
    @scores = scores
  end

  def latest
    @scores.last
  end

  def personal_best
    @scores.max
  end

  def personal_top_three
    @scores.sort.reverse.first 3
  end
end

File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private IList<int> scores;

    public HighScores(IList<int> list)
    {
        scores = list;
    }

    public IList<int> Scores() => this.scores;

    public int Latest() => this.scores.Last();

    public int PersonalBest() => this.scores.Max();

    public IList<int> PersonalTopThree() => this.scores
            .OrderByDescending(s => s)
            .Take(3)
            .ToList();
}
